org.eclipse.gef/src/org/eclipse/gef/internal/ui/palette/editparts/DrawerEditPart.java

```diff
@@ -99,6 +100,11 @@ public <T> T getAdapter(final Class<T> key) {
 		if (key == MouseWheelHelper.class) {
 			return key.cast(new ViewportMouseWheelHelper(this));
 		}
+		if (key == Clickable.class) {
+			DrawerFigure drawerFigure = getFigure();
+			Clickable drawerToggle = drawerFigure.getCollapseToggle();
+			return key.cast(drawerToggle);
+		}
```

Comment:
I'm not entirely sure whether it's a good idea to adapt the `DrawerEditPart` to a `Clickable`, or only the `DrawerFigure`. The latter means more work, as the `DrawerFigure` then needs to implement the `IAdaptable` interface.

And given that all of these are internal classes, you can't access them directly but instead have to go via the `EditPart` and `IFigure` interfaces.